### PR TITLE
Fix/attachmentlist + refactor

### PR DESCRIPTION
**Pull request by lassopicasso**
## Description

The attachmentList component has undergone a simplification refactor, thx @olemartinorg! 

In response to feedback received in a [previous PR](https://github.com/Altinn/app-frontend-react/issues/1489), it was identified that the variants `dataTypeIds=[]` and `["include-all"]` were not functioning optimally regarding to process task. The current functionality was to include dataTypes from only current task, this has been changed to include all process tasks. 
Based on these changes a new variant is implemented, `dataTypeIds=["from-task"]`. This variant operates similarly to an empty or undefined array but exclusively includes dataTypes from the current task.

Both unit and Cypress tests have been updated because of these changes. Furthermore, to resolve WCAG-related issues in Cypress related to `attachmentLists`, the material-ui in AltinnAttachment has been replaced with the List component from the Designsystem.

Updated documentation is located here: https://github.com/Altinn/altinn-studio-docs/commit/382b5aa5f4af054656629f494b9dbe1a6630f8d3

*There may be some additional changes to the styling of `altinnAttachment` following tomorrow's (21.11) design meeting with @olavflar.*

## Related Issue(s)

- closes #{issue number}

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [x] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/components/atoms/AltinnAttachment.tsx [hunk 1]
```diff
@@ -1,138 +1,49 @@
...
-            ))}
-      </List>
-    </>
+    <List
+      id={id}
+      heading={lang(title)}
+      data-testid='attachment-list'
+      className={classes.attachmentList}
+    >
+      {filteredAndSortedAttachments &&
```
Comment (mikaelrss) [inline src/components/atoms/AltinnAttachment.tsx:31]: You could use optional chaining here `filteredAndSortedAttachments?.map(...`

File: src/components/atoms/AltinnAttachment.tsx [hunk 2]
```diff
@@ -1,138 +1,49 @@
...
-}
+export function AltinnAttachment({ attachments, id, title }: IAltinnAttachmentProps) {
+  const { lang, langAsString, selectedLanguage } = useLanguage();
 
-export function AltinnAttachment({ attachments, listDisableVerticalPadding, nested, id }: IAltinnAttachmentProps) {
-  const classes = useStyles();
-  const { selectedLanguage } = useLanguage();
+  const filteredAndSortedAttachments = attachments
+    ?.filter((attachment) => attachment.name)
+    .sort((a, b) => (a.name && b.name ? a.name.localeCompare(b.name, selectedLanguage) : 0));
```
Comment (mikaelrss) [inline src/components/atoms/AltinnAttachment.tsx:22]: `localeCompare` won't sort by numbers if not specifying the `{ numeric: true }` option. So `"1" <  "2"` but `"2" !< "10"`. I'm not sure if that's relevant in the case of attachmentlist, but you could imagine a user uploading files that start with numeric values :sweat_smile:

**Review by mikaelrss — APPROVED** [review]
Looks good! :clap: I had a couple of nit picky comments. You can decide if you want to change anything :sweat_smile:

File: src/components/atoms/AltinnAttachment.tsx (hunk 2 above)
Comment (lassopicasso) [reply src/components/atoms/AltinnAttachment.tsx:22]: Aha :D Thx, will check it out.

File: src/components/atoms/AltinnAttachment.tsx (hunk 1 above)
Comment (lassopicasso) [reply src/components/atoms/AltinnAttachment.tsx:31]: Will do, thx!

File: test/e2e/support/customReceipt.ts [hunk 3]
```diff
@@ -8,8 +20,18 @@ export const customReceipt: ILayout = [
     type: 'Paragraph',
     textResourceBindings: { title: 'Takk for din innsending, dette er en veldig fin custom kvittering.' },
   },
-  { id: 'r-header-pdfs', type: 'Header', size: 'M', textResourceBindings: { title: 'receipt.title_submitted' } },
+  { id: 'r-pdfs-header', type: 'Header', size: 'M', textResourceBindings: { title: 'receipt.title_submitted' } },
   { id: 'r-pdfs', type: 'AttachmentList', dataTypeIds: ['ref-data-as-pdf'] },
```
Comment (olemartinorg) [inline test/e2e/support/customReceipt.ts:24]: Whoops, ser at denne ble duplikat - jeg la til en "Bare PDF-er" under også

**Review by olemartinorg — APPROVED** [review]

**Pull request merged**